### PR TITLE
Hotfix/fetcher

### PR DESCRIPTION
**Pull request by dcyou**
Hi,
Not sure but should solve the problem of revalidate / mutate by keeping the default fetcher defined in the `createSWR` conf
In my case I am using axios but when using rev or mut it use fetch API

image

### CODE REVIEW

**Comment by ConsoleTVs** [comment]
You're right, but it is also intended.

The `revalidate` and `mutate` are not hooks, meaning you can call them whenever you want. and the inject() can only be called inside setup().

We also have to take into consideration that your fix does not take into account that those methods are not 100% compatible with what inject(vswr) does provide. You can see that for example, the revalidation options only takes in the fetcher and the dedupingInterval (https://github.com/ConsoleTVs/vswr/blob/fd156e277eb0f060df80d8434ef19ea86b1529e3/src/api/revalidate.ts#L11) 

Nevertheless, I am aware this causes confusion, so here's what I am going to do:

1. For you, you can currently abstract the fetcher into something like this:

```js
export const commonOptions = {
  fetcher: (key) => axios.get(key).then(res => res.data)
}
```

Then you can use those common options in the createSWR and also the revalidate and mutate functions as such:

```js
export const vswr = createSWR({
  ...commonOptions,
  // Other options
});
```

and you can also pass what you want into the functions as:

```js
revalidate(key, { ...commonOptions, /* Other options...*/ })
mutate(key, value, { revalidateOptions: { ...commonOptions, /* Other options...*/ } })
```

2. For the matter, I do agree this the behaviour you describe is what should happen when possible. The happy part of this is that I do have a working framework-agnostic version of vswr at https://github.com/ConsoleTVs/swrev and I can implement this library using a few lines and that framework agnostic implementation. Once I do this, I will also add the feature described here and allow using the fetcher (among any other repeated options) on all the functions.

TLDR; I will be adding this as part of a rework of this package (internally only! - same API), meanwhile I've provided you with a simple fix (although a bit tedious to keep importing those options).

**Comment by ConsoleTVs** [comment]
As for the estimated time for this feature, I expect to have it working this weekend, but it's my free time so I won't promise anything (since it also needs some testing).

**Comment by ConsoleTVs** [comment]
Alright, the new version of `vswr` (1.0.0) has been released with this issue fixed. Feel free to open any other issue / PR if there are still errors somewhere. Please read the upgrade guide at: https://github.com/ConsoleTVs/vswr#upgrade-guide

Have a nice day.

**Comment by dcyou** [comment]
ok both solutions work thanks
